### PR TITLE
Refactor and add specs for Transformation - AssessTransformation

### PR DESCRIPTION
**Pull request by fdupont-redhat**
Now that [PR#391](https://github.com/ManageIQ/manageiq-content/pull/391) is merged, this PR refactors the transformation assessment to split the code in max-5-lines-ish methods and adds the associated tests.

Associated RHBZ: https://bugzilla.redhat.com/show_bug.cgi?id=1600870

### CODE REVIEW

**Comment by fdupont-redhat** [comment]
@miq-bot add-label transformation, bug, gaprindashvili/yes, blocker
@miq-bot add-reviewer mkanoor
@miq-bot remove-label wip

**Comment by fdupont-redhat** [comment]
@miq-bot remove-label bug
@miq-bot add-label technical debt, refactoring

File: content/automate/ManageIQ/Transformation/Common.class/__methods__/assesstransformation.rb [hunk 1]
```diff
@@ -5,19 +5,18 @@ module Common
         class AssessTransformation
           SUPPORTED_SOURCE_EMS_TYPES = ['vmwarews'].freeze
           SUPPORTED_DESTINATION_EMS_TYPES = ['rhevm'].freeze
-          REQUIRED_CUSTOM_ATTRIBUTES = {
-            'rhevm' => %i(rhv_export_domain_id rhv_cluster_id rhv_storage_domain_id)
-          }.freeze
 
           def initialize(handle = $evm)
             @handle = handle
+            @task ||= ManageIQ::Automate::Transformation::Common::Utils.task(@handle)
```
Comment (mkanoor) [inline content/automate/ManageIQ/Transformation/Common.class/__methods__/assesstransformation.rb]: @fdupont-redhat The initialize is called once per instance so you dont need the ||= it should just be
```
@task = ManageIQ::Automate::Transformation::Common::Utils.task(@handle)
```
also for source_vm

Comment (fdupont-redhat) [reply content/automate/ManageIQ/Transformation/Common.class/__methods__/assesstransformation.rb]: Good catch.

**Comment by simaishi** [comment]
As per BZ Target Release, removing `gaprindashvili/yes`

**Pull request merged**